R/undesirable_function_linter.R

```diff
@@ -14,27 +14,27 @@ undesirable_function_linter <- function(fun = default_undesirable_functions,
                                         symbol_is_undesirable = TRUE) {
   stopifnot(is.logical(symbol_is_undesirable))
 
+  if (symbol_is_undesirable) {
+    tokens <- c("SYMBOL_FUNCTION_CALL", "SYMBOL")
+  } else {
+    tokens <- "SYMBOL_FUNCTION_CALL"
+  }
+
+  xp_condition <- xp_and(
+    xp_text_in_table(names(fun)),
+    paste0(
+      "not(parent::expr/preceding-sibling::expr[SYMBOL_FUNCTION_CALL[",
+      xp_text_in_table(c("library", "require")),
+      "]])"
+    ),
+    "not(preceding-sibling::OP-DOLLAR)"
+  )
+  xpath <- paste(glue::glue("//{tokens}[{xp_condition}]"), collapse = " | ")
```

Comment:
WDYT about being more explicit instead of using collapse?

```
if (symbol_is_undesirable) {
  xpath <- glue::glue("//SYMBOL_FUNCTION_CALL[{xp_condition}] | //SYMBOL[{xp_condition}]")
} else {
  xpath <- glue::glue("//SYMBOL_FUNCTION_CALL[{xp_condition}]
}
```

or

```
xpath <- glue::glue("//SYMBOL_FUNCTION_CALL[{xp_condition}]"
if (symbol_is_undesirable) {
  xpath <- paste(xpath, glue::glue("| //SYMBOL[{xp_condition}]"))
}
```

?

Reply:
I like the first suggestion